src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -437,6 +441,15 @@ private static GeneratorState SetGeneratorException(CommonMessageProvider provid
             return new GeneratorState(generatorState.Info, e, diagnostic);
         }
 
+        private static string GetDirectoryForGenerator(Compilation compilation, ISourceGenerator generator)
```

Comment:
Rather than take in the `Compilation` here I would just take `string generatedFilesOutputDirectory` instead. Taking the `Compilation` makes this method very "heavy" where in theory it's just a path manipulation function, doesn't need a `Compilation`